plutus-ledger/src/Ledger/Tx/CardanoAPI.hs

```diff
@@ -423,6 +424,21 @@ toCardanoTxBodyContent P.Params{P.pProtocolParams, P.pNetworkId} sigs P.Tx{..} =
         , txUpdateProposal = C.TxUpdateProposalNone
         }
 
+  where
+    -- FIXME what do we need to do here if
```

Comment:
:facepalm:

> FIXME what do we need to do here if the `P.Tx` refers to a datum that is not in txData? Do we need to fail with a `ToCardanoError` or we can include the datum hash in the outputs anyway? Here I am just reproducing the previous behaviour which is include the hash anyway.

Reply:
I moved `lookupDatum` here because it's an helper function to `toCardanoTxBodyContent`